plugins/CorePluginsAdmin/CorePluginsAdmin.php

```diff
@@ -26,10 +29,35 @@ public function registerEvents()
             'AssetManager.getStylesheetFiles'        => 'getStylesheetFiles',
             'System.addSystemSummaryItems'           => 'addSystemSummaryItems',
             'Translate.getClientSideTranslationKeys' => 'getClientSideTranslationKeys',
-            'PluginManager.pluginActivated'          => 'onPluginActivated'
+            'PluginManager.pluginActivated'          => 'onPluginActivated',
+            'PluginManager.pluginInstalled'          => 'addPluginChanges',
+            'Updater.componentUpdated'               => 'addPluginChanges',
+            'PluginManager.pluginUninstalled'        => 'removePluginChanges'
         );
     }
 
+    /**
+     * Add any changes from newly installed or updated plugins to the changes table
+     *
+     * @param string $pluginName The name of the plugin that was updated or installed
+     */
+    public function addPluginChanges(string $pluginName)
+    {
+        $changes = new ChangesModel(Db::get(), PluginManager::getInstance());
+        $changes->addChanges($pluginName);
+    }
+
+    /**
+     * Remove any changes from a plugin that has been uninstalled
+     *
+     * @param string $pluginName The name of the plugin that was uninstalled
+     */
+    public function removePluginChanges(string $pluginName)
+    {
+        $changes = new ChangesModel(Db::get(), PluginManager::getInstance());
+        $changes->removeChanges($pluginName);
+    }
```

Comment:
We maybe should add a try/catch around adding/removing the changes. I'm actually not sure if that could otherwise cause issues when someone updates from an older version to a version that already contains some plugins with changes files. Depending on the order the updates are processed maybe the plugins are updated before the database migration runs and adding plugin changes might fail, as the table not yet exists.

Reply:
I've added a try/catch to both `core/Change/Model::removeChanges()` and `core/Change/Model::addChange()` to handle missing table exceptions.